overlord/devicestate/devicestate.go

```diff
@@ -95,23 +95,25 @@ func canAutoRefresh(st *state.State) (bool, error) {
 		return false, nil
 	}
 
-	_, err := Model(st)
-	if err == state.ErrNoState {
-		// no model, no need to wait for a serial
-		// can happen only on classic
-		return true, nil
-	}
-	if err != nil {
-		return false, err
-	}
-
+	// we now try to get a serial everywhere, core and classic
+	// and even if there's no seeded model, using a dummy model
+	// in that case.
```

Comment:
The comment seems a bit strange here. It seems to justify the lack of the code that is not here anymore. :)